cmd/lazygpt/app/chat.go

```diff
@@ -3,18 +3,85 @@
 package app
 
 import (
+	"context"
+	"errors"
 	"fmt"
 	"os"
 	"strings"
 
-	"github.com/c-bata/go-prompt"
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/AlecAivazis/survey/v2/terminal"
 	"github.com/spf13/cobra"
 
 	"github.com/lazygpt/lazygpt/pkg/plugin"
 	"github.com/lazygpt/lazygpt/plugin/api"
 	"github.com/lazygpt/lazygpt/plugin/log"
 )
 
+const (
+	ErrFailedSurvey    = "failed to ask survey"
+	ErrPromptNotString = "prompt must be a string"
+	ErrPromptEmpty     = "prompt can not be empty"
+	ErrMessagesNil     = "messages can not be nil"
+)
+
+type promptExecutor func(context.Context, string) string
+
+// promptValidator is a basic validator for the survey prompt.
+func promptValidator(in interface{}) error {
+	input, ok := in.(string)
+	if !ok {
+		return errors.New(ErrPromptNotString)
```

Comment:
Stylistic, I'd recommend making the errors vars (`var ErrPromptNotString = errors.New("prompt must be a string")`) at the global level, instead of dynamic creation in the functions. That allows for using `errors.Is` / `errors.As` for error handling down the road (for example if we move the prompting into a another package).